aioconsole/events.py

```diff
@@ -78,8 +80,26 @@ def set_interactive_policy(*, locals=None, banner=None, serve=None,
         prompt_control=prompt_control)
     asyncio.set_event_loop_policy(policy)
 
+def exec_pythonstartup(locals):
+    filename = os.environ.get('PYTHONSTARTUP')
+    if filename:
+        if os.path.isfile(filename):
+            with open(filename) as fobj:
+                startup_file = fobj.read()
+            try:
+                exec(startup_file, globals(), locals)
+            except Exception as e:  # pragma: no cover
+                tb = traceback.format_exc()
+                print(tb)
+
+        else:
+            print('Could not open PYTHONSTARTUP - No such file: %s', filename)
 
 def run_console(*, locals=None, banner=None, serve=None, prompt_control=None):
+    if locals is None:
+        locals = {}
+    exec_pythonstartup(locals)
+
```

Comment:
I think this should go to the [run_apython](https://github.com/vxgmichel/aioconsole/blob/c5d2d23a5373deab561d738548e16c10ccc3b979/aioconsole/apython.py#L82) function instead. Any objection?

Reply:
There is actually a more fundamental question of whether or not we want the code exec'd inside the context of the ioloop or not.  There is actually an argument to be made for actually pushing this down into the AsynchronousConsole, taking advantage of the built in compile/aexec work so the locals of the PYTHONSTARTUP contains the built-in items which are overridden (namely print).  That implementation looks like this: https://github.com/vxgmichel/aioconsole/compare/master...vEpiphyte:feature_pythonstartup_inloop?expand=1

Or it could be pushed up to apython, right before events.runconsole() is executed to pre-populate locals.   An implementation of that can be seen here https://github.com/vxgmichel/aioconsole/compare/master...vEpiphyte:feature_pythonstartup_apython?expand=1

Reply:
Interesting! It's a nice feature to be able to bootstrap an interactive session with some user code, but I don't think that this should be `PYTHONSTARTUP` job. For instance [code.interact](https://docs.python.org/3.5/library/code.html#code.interact) does not use it, and I think we should stick to the standard behavior.

I can see two ways of dealing with this:
- Either we move this logic to `apython` as you suggested [here](https://github.com/vxgmichel/aioconsole/compare/master...vEpiphyte:feature_pythonstartup_apython?expand=1)
- Or we implement some optional bootstrap logic in the asynchronous console (similar to what you suggested [here](https://github.com/vxgmichel/aioconsole/compare/master...vEpiphyte:feature_pythonstartup_inloop?expand=1)).  Then we can change `run_apython` to pass the content of the `PYTHONSTARTUP` as bootstrap to `events.runconsole()`.

For the moment I would go with the first approach unless you have a use case for the second one :)

Reply:
I don't have a strong use-case for the second situation. I'll merge the apython solution into this branch and master.